homeassistant/components/rainbird/__init__.py

```diff
@@ -84,36 +92,46 @@
 )
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Rain Bird component."""
 
     hass.data[DATA_RAINBIRD] = []
-    success = False
-    for controller_config in config[DOMAIN]:
-        success = success or _setup_controller(hass, controller_config, config)
 
-    return success
+    tasks = []
+    for controller_config in config[DOMAIN]:
+        tasks.append(_setup_controller(hass, controller_config, config))
+    return all(await asyncio.gather(*tasks))
```

Comment:
```suggestion
    return all(
        await asyncio.gather(
            _setup_controller(hass, controller_config, config)
            for controller_config in config[DOMAIN]
        )
    )
```

Not sure its better but this probably works as well (I have no way to test)

Reply:
I somehow dropped this suggestion, will re-apply.